From 961c8cbd2a489277876aeffa40ebdee5eae29f1f Mon Sep 17 00:00:00 2001
From: Dipankar Bhardwaj <dipankarb@google.com>
Date: Tue, 25 Jul 2023 17:42:15 +0000
Subject: [PATCH] Expand the size of zero_addr to the maximum size of a single
 request.

Bug: 285078221
Bug: 284989074
Test: atest RedactUriDeviceTest
(cherry picked from commit e45ef2e4c41c94b930c7c385b15f5756e0d24a81)
DISABLE_TOPIC_PROTECTOR
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:7c3271e6fe19f33ce82ea96397a1192adc1614d4)
Merged-In: Ibf9f024a8d0bd5c3b3f2c7509f27adeb5077aa5e
Change-Id: Ibf9f024a8d0bd5c3b3f2c7509f27adeb5077aa5e
---
 jni/FuseDaemon.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/jni/FuseDaemon.cpp b/jni/FuseDaemon.cpp
index a0eb54754..3c585ce61 100644
--- a/jni/FuseDaemon.cpp
+++ b/jni/FuseDaemon.cpp
@@ -103,7 +103,9 @@ const bool IS_OS_DEBUGABLE = android::base::GetIntProperty("ro.debuggable", 0);
 // Stolen from: android_filesystem_config.h
 #define AID_APP_START 10000
 
-constexpr size_t MAX_READ_SIZE = 128 * 1024;
+#define FUSE_MAX_MAX_PAGES 256
+
+const size_t MAX_READ_SIZE = FUSE_MAX_MAX_PAGES * getpagesize();
 // Stolen from: UserHandle#getUserId
 constexpr int PER_USER_RANGE = 100000;
 
-- 
2.48.1.262.g85cc9f2d1e-goog

